blosc2/ndarray.py

```diff
@@ -168,6 +169,32 @@ def __setitem__(self, key, value):
 
         return super(NDArray, self).set_slice(key, value)
 
+    def iterchunks_info(self):
+        """
+        Iterate over :paramref:`self` chunks, providing info on index and special values.
+
+        Yields
+        ------
+        info: namedtuple
+            A namedtuple with the following fields:
+            nchunk: the index of the chunk.
+            coords: the coordinates where the chunk starts.
+            cratio: the compression ratio of the chunk.
+            special: the special value enum of the chunk; if 0, the chunk is not special.
+            repeated_value: the repeated value for the chunk; if not SpecialValue.VALUE, it is None.
+            lazychunk: a buffer with the complete lazy chunk.
+        """
+        ChunkInfoNDArray = namedtuple(
+            "ChunkInfoNDArray", ["nchunk", "coords", "cratio", "special", "repeated_value", "lazychunk"]
+        )
+        chunks_idx = np.array(self.shape) // np.array(self.chunks)
```

Comment:
This should be` np.array(self.ext_shape) // np.array(self.ext_chunks)` in case there's padding.

Reply:
Thanks.  Fixed in https://github.com/Blosc/python-blosc2/pull/119/commits/bdd16148c7b666db5cea11c323d6cf199d16d4fc